src/encrypt.ts

```diff
@@ -0,0 +1,57 @@
+import * as crypto from "crypto";
+import * as t from "io-ts";
+
+const IV_LENGTH = 16;
+const AES_LENGTH = 16;
+
+const EncryptTuple = t.interface({
+  // Hybrid encrypted result (Base64)
+  encryptedOutput: t.string,
+
+  // random AES string (Base64)
+  iv: t.string,
+
+  // AES Key encrypted with RSA public key (Base64)
+  rsaAesKey: t.string
+});
+
+export type EncryptTuple = t.TypeOf<typeof EncryptTuple>;
+
+export function hybridEncrypt(input: string, rsaPubKey: string): EncryptTuple {
+  const iv = crypto.randomBytes(IV_LENGTH);
+  const aesKey = crypto.randomBytes(AES_LENGTH);
+  const cipher = crypto.createCipheriv("aes-128-cbc", Buffer.from(aesKey), iv);
+  const encryptedOutput = Buffer.concat([
+    cipher.update(input),
+    cipher.final()
+  ]).toString("base64");
+  const rsaAesKey = crypto
+    .publicEncrypt(rsaPubKey, Buffer.from(aesKey))
+    .toString("base64");
+  return {
+    encryptedOutput,
+    iv: iv.toString("base64"),
+    rsaAesKey
+  } as EncryptTuple;
+}
+
+export function hybridDecrypt(
```

Comment:
If I use ecrypt or decrypt function with wrong keys pair, I will get an exception.
For example this code raise an exception:

```typescript
const encryptedPayload = toEncryptedPayload(rsaPublicKey, aTextToEncrypt);
    expect(encryptedPayload).toHaveProperty("encryptedPayload");
    const decryptedString = fromEncryptedPayload(
      rsaPrivateKey + "x", //wrong private key
      encryptedPayload
    );
```

Could this implementation be more error friendly?

```typescript
export function fromEncryptedPayload(
  rsaPrivateKey: string,
  encryptedPayload: EncryptedPayload
): Either<Error, string> {
  return tryCatch(
    () => {
      const iv = Buffer.from(encryptedPayload.iv, "base64");
      const aesKey = crypto.privateDecrypt(
        rsaPrivateKey,
        Buffer.from(encryptedPayload.encryptedKey, "base64")
      );
      const decipher = crypto.createDecipheriv(
        "aes-128-cbc",
        Buffer.from(aesKey),
        iv
      );
      const decrypted = decipher.update(
        Buffer.from(encryptedPayload.encryptedPayload, "base64")
      );
      const output = Buffer.concat([decrypted, decipher.final()]);
      return output.toString("utf-8");
    },
    (e: unknown) => (e instanceof Error ? e : new Error(String(e)))
  );
}
```

Reply:
that's a very good point, probably yes give me a moment :)

Reply:
fixed